prql-compiler/src/lib.rs

```diff
@@ -143,6 +150,27 @@ impl Target {
     }
 }
 
+impl FromStr for Target {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Target, Self::Err> {
+        // We have a closure here because we can't create the error in the
+        // pipeline, since it needs to be in two places, and we'd need to clone.
+        // (Though possibly it's too optimize-y.)
+        let not_found_error = |s| {
+            Error::new(Reason::NotFound {
+                name: format!("{s:?}"),
+                namespace: "target".to_string(),
+            })
+        };
+        s.strip_prefix("sql.")
```

Comment:
@eitsupi this wasn't actually as simple as I thought. I made a small tweak to remove the extra condition. It's fine to revert if you thought the existing one was clearer.

Whenever you're happy, let's merge!

Reply:
Thank you for the update. I will merge this.